src/Features/Core/Portable/SignatureHelp/SignatureHelpService.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.SignatureHelp
+{
+    /// <summary>
+    /// A per language service for constructing context dependent list of signatures that 
+    /// can be presented to a user while typing in an editor.
+    /// </summary>
+    internal abstract class SignatureHelpService : ILanguageService
+    {
+        /// <summary>
+        /// Gets the service corresponding to the specified document.
+        /// </summary>
+        public static SignatureHelpService GetService(Document document)
+        {
+            return document.Project.LanguageServices.GetService<SignatureHelpService>();
```

Comment:
This is intended to be a public API like we have for `CompletionService.GetService(Document)`.